### DISCUSSION TITLE
Coral for Frigate in Docker on Ubuntu

### DISCUSSION THREAD

**Discussion by tlundste** [blakeblackshear/frigate#1746] [Ask A Question]
Running a NUC8i3, using Ubuntu, with Frigate running in a Docker image.  I see on the Coral website, their products are supported for Debian... will it work with Ubuntu (and the rest of my setup)?
I plan on getting the USB device, if it does work, as what I am reading the less expensive M.2 device might not work in a NUC, because the M.2 slot is configured for storage only (if anyone has experience to the contrary, I'd like to know how it worked for you).

**Comment by mitchross** [comment]
https://github.com/google-coral/edgetpu/issues/256#issuecomment-920839050

Save your self the hassle and just get USB

**Reply by haggercody** [reply]
yea the coral usb works with almost zero effort on my unbuntu so i'd say it is safe.  I have way more trouble with my headphones than this thing.

**Comment by shred86** [comment]
I'm using an M.2 Coral EdgeTPU on my Mini PC and the setup was simple. Followed the first [two steps](https://coral.ai/docs/m2/get-started/#requirements) on the Coral website and pass it to the docker run command (`--device /dev/apex_0:/dev/apex_0`). As long as your M.2 slot uses the PCI-E interface, it should work fine (just make sure it's PCI-E and not SATA).
